pandas/tests/series/test_indexing.py

```diff
@@ -287,6 +287,12 @@ def test_getitem_generator(self):
         assert_series_equal(result, expected)
         assert_series_equal(result2, expected)
 
+    def test_type_promotion(self):
+        # GH12599
+        df1 = pd.Series()
+        df1["a"] = pd.Timestamp("2016-01-01")
+        df1["b"] = 3.0
+
```

Comment:
well I would like to have a comparision for the expected on ``df1``. And since its a series, call it ``s`` (and not df1).